libpod/container.go

```diff
@@ -102,6 +102,20 @@ func (ns LinuxNS) String() string {
 	}
 }
 
+// Valid restart policy types.
+const (
+	// RestartPolicyNone indicates that no restart policy has been requested
+	// by a container.
+	RestartPolicyNone = ""
+	// RestartPolicyNo is identical in function to RestartPolicyNone.
+	RestartPolicyNo = "no"
+	// RestartPolicyAlways unconditionally restarts the container.
+	RestartPolicyAlways = "always"
+	// RestartPolicyOnFailure restarts the container on non-0 exit code,
+	// with an optional maximum number of retries.
+	RestartPolicyOnFailure = "on-failure"
```

Comment:
```
    -   **RestartPolicy** – The behavior to apply when the container exits.  The
            value is an object with a `Name` property of either `"always"` to
            always restart, `"unless-stopped"` to restart always except when
            user has manually stopped the container or `"on-failure"` to restart only when the container
            exit code is non-zero.  If `on-failure` is used, `MaximumRetryCount`
            controls the number of times to retry before giving up.
            The default is not to restart. (optional)
            An ever increasing delay (double the previous delay, starting at 100mS)
            is added before each restart to prevent flooding the server.
```

Reply:
Unless-stopped is very much a "restarted the daemon" thing, so I just throw an error about it in pkg/spec if they try to pass it.

We still want to point people to systemd for cases like that, where the container will be restarted after system reboot, for example. The manpages make this clear, but it might be good to add the warning to Podman itself too.

Reply:
My reading of unless-stopped, seems to be exactly what you are designing.  IE If a user stops a container then it will not restart, otherwise if the container fails, it will be restarted.

Reply:
It seems the interpretation depends on where we look. The man page is referring to the daemon start only while the docs state:

> Similar to always, except that when the container is stopped (manually or otherwise), it is not restarted even after Docker daemon restarts

This makes me believe that we can support it.

Reply:
From that description, it's identical to `always` - restart policy never triggers after the container is stopped via an API call.

Reply:
(Minus the daemon restart aspect)